client/cli/src/config.rs

```diff
@@ -24,10 +24,11 @@ use crate::{
 use app_dirs::{AppDataType, AppInfo};
 use names::{Generator, Name};
 use sc_service::config::{
-	Configuration, DatabaseConfig, ExecutionStrategies, ExtTransport, KeystoreConfig,
-	NetworkConfiguration, NodeKeyConfig, PrometheusConfig, PruningMode, Role, TelemetryEndpoints,
-	TransactionPoolOptions, WasmExecutionMethod,
+	Configuration, DatabaseConfig, ExtTransport, KeystoreConfig, NetworkConfiguration,
+	NodeKeyConfig, PrometheusConfig, PruningMode, TelemetryEndpoints,
+	TransactionPoolOptions, WasmExecutionMethod, Role,
 };
+use sc_client_api::execution_extensions::ExecutionStrategies;
```

Comment:
Can you re-export ExecutionStrategies from sc_service? Otherwise any application that will want to use this method will need to add sc-client-api in its dependencies

Reply:
hmm `sc-service` is a pretty heavy thing to depend on for struct that doesn't originate within it. That's just how i see it.

>  Otherwise any application that will want to override this method will need to add sc-client-api in its dependencies. 

Is there a problem with this?

Reply:
In the end everything is linked together so even if something doesn't appear in the dependency it will be used by another dependency somewhere.

> Is there a problem with this?

I'm not sure yet tbh. I don't very well understand our crate and module separation. The only thing I see (from a cli perspective) is that having more modules in this trait increases the number of modules required to do a cli. So yes/no, maybe, I really don't know.

btw: I just checked the code and ExecutionStrategies was originally re-exported by sc-service. Is there a reason why you removed that?

Reply:
> btw: I just checked the code and ExecutionStrategies was originally re-exported by sc-service. Is there a reason why you removed that?

Primarily because during refactorings, I changed everywhere it was imported from `sc_serivce` to `sc_client_api`. 

> The only thing I see (from a cli perspective) is that having more modules in this trait increases the number of modules required to do a cli.

Hmm, I see what you mean then, for now, it should be exported from `sc_service`.

Reply:
in general, the approach we are heading for is to have fewer and fewer crates depend on `sc-service` – which is a gigantic piece to compile – but rather have `sc-client-api` the source of all `traits` to be used. Reasoning here is simple: the `sc-client-api` should effectively be a thin API description similar as our primitives would be and while `sc-service` is the top-most crate that plugs them all together, you should only need the `sc-client-api` in any crate below. Thus, this should be imported directly from `sc-client-api` and not be re-exported from `sc-service`, because eventually very few things should directly depend on `sc-service`.

And once they do not depend on`sc-service` but rather that thing `sc-client-api`, a run of `cargo check` or `cargo test` on those crates directly would be vastly faster than if `sc-service` has to be compiled as well.

Reply:
When you build an application you need sc-cli to make the cli part but you also need sc-service for the builder, ChainSpec and ChainType. Do you plan to also move those away so building an application won't require sc-service at all?

Reply:
@cecton it isn't high priority, but yes, ideally `sc-cli` is generic enough that it wouldn't need `sc-service` but works with anything that implements the same interfaces. The thinking is that we might want to have different internal `clients`/`services` that plugs everything together (e.g. an in-memory-direct-executing-client for testing) but the outer shouldn't care – same as for runtime itself.

so whenever we touch the code base, we do want to _slowly_ move into that direction ;) .

Reply:
I trust you :slightly_smiling_face: